### ISSUE TITLE
Lab 3 CI, review

### ISSUE THREAD

**Issue by LorenzoUgo** [AnjaliVaghjiani/CI2024_lab3#2]
The choice of a BI-directional A* alghoritm is nice to speed up the exploration.
The solution should theoretically take _t/2_ (with t = time for an unidirectional search), that is quite interesting. But, you are not decreasing the number of state explored (be careful) .
Moreover, with this kind of implementation, where the code is not parallelize, does not benefit too much from your idea: the two cycle for the exploration are put in a sequential mode, not in parallel.

To sum up: the idea is nice, but how the code is written does not explore the full potential of it.
